step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/AutomationPackageExecutor.java

```diff
@@ -94,37 +94,42 @@ public List<String> runInIsolation(InputStream apInputStream, String inputStream
-
                 if ((planFilter == null || planFilter.isSelected(plan)) && plan.getRoot().getClass().getAnnotation(Artefact.class).validForStandaloneExecution()) {
-                    ExecutionParameters params = parameters.toExecutionParameters();
-
-                    HashMap<String, String> repositoryParameters = new HashMap<>();
-
-                    // save apName + contextId + planName to support re-execution
-                    repositoryParameters.put(IsolatedAutomationPackageRepository.AP_NAME, apName);
-                    repositoryParameters.put(IsolatedAutomationPackageRepository.REPOSITORY_PARAM_CONTEXTID, contextId.toString());
-                    repositoryParameters.put(IsolatedAutomationPackageRepository.PLAN_NAME, plan.getAttribute(AbstractOrganizableObject.NAME));
-
-                    // store the reference from original repository object
-                    if (parameters.getOriginalRepositoryObject() != null) {
-                        repositoryParameters.putAll(parameters.getOriginalRepositoryObject().getRepositoryParameters());
-                    }
-
-                    params.setRepositoryObject(new RepositoryObjectReference(repoId, repositoryParameters));
-                    params.setDescription(CommonExecutionParameters.defaultDescription(plan));
-
-                    // for instance, set the project for multitenant application
-                    if (objectEnricher != null) {
-                        objectEnricher.accept(params);
-                    }
+                    applicablePlans.add(plan);
+                } else {
+                    somePlansFiltered = true;
+                }
+            }
 
+            if (parameters.getWrapIntoTestSet() == null || !parameters.getWrapIntoTestSet()) {
+                // run each plans in separate execution (apply the plan name filter to use the single file in execution)
+                for (Plan plan : applicablePlans) {
+                    ExecutionParameters params = prepareExecutionParams(
+                            parameters, objectEnricher, apName,
+                            contextId, repoId,
+                            CommonExecutionParameters.defaultDescription(plan), plan.getAttribute(AbstractOrganizableObject.NAME)
+                    );
                     String newExecutionId = this.scheduler.execute(params);
                     if (newExecutionId != null) {
                         executions.add(newExecutionId);
                     }
                 }
+            } else {
+                // wrap all plans in test set
+                ExecutionParameters params = prepareExecutionParams(
+                        parameters, objectEnricher, apName,
+                        contextId, repoId,
+                        null,
+                        somePlansFiltered ? applicablePlans.stream().map(p -> p.getAttribute(AbstractOrganizableObject.NAME)).collect(Collectors.joining(",")) : null
```

Comment:
The open question here is how to handle the case when we need to wrap plans and at the same time we have some filter defined in `AutomationPackageExecutionParameters.planFilter`. Current solution applies this filter to each `Plan` in automation package and adds the `PARAM_INCLUDE_PLANS` to the repository parameters. Otherwise inside the `IsolatedAutomationPackageRepository` we cannot filter out the excluded plans.

Do you see some more pretty way to solve this?

Reply:
@iegorov777 I'll create a follow up task to discuss this